src/ducks/sep6WithdrawAsset.ts

```diff
@@ -0,0 +1,328 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "config/store";
+import { accountSelector } from "ducks/account";
+import { settingsSelector } from "ducks/settings";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { log } from "helpers/log";
+import { checkDepositWithdrawInfo } from "methods/checkDepositWithdrawInfo";
+import { programmaticWithdrawFlow } from "methods/sep6";
+import {
+  sep10AuthStart,
+  sep10AuthSign,
+  sep10AuthSend,
+} from "methods/sep10Auth";
+import { collectSep12Fields, putSep12FieldsRequest } from "methods/sep12";
+import { checkTomlForFields } from "methods/checkTomlForFields";
+import {
+  Asset,
+  ActionStatus,
+  Sep6WithdrawAssetInitialState,
+  Sep6WithdrawResponse,
+  RejectMessage,
+  TomlFields,
+  CheckInfoType,
+  AnyObject,
+} from "types/types.d";
+
+export const initiateWithdrawAction = createAsyncThunk<
+  { fields: AnyObject; status: ActionStatus },
+  Asset,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep6WithdrawAsset/initiateWithdrawAction",
+  async (asset, { rejectWithValue, getState }) => {
+    const { assetCode, assetIssuer, homeDomain } = asset;
+    const { data, secretKey } = accountSelector(getState());
+    const { pubnet } = settingsSelector(getState());
+    const networkConfig = getNetworkConfig(pubnet);
+    const publicKey = data?.id;
+
+    // This is unlikely
+    if (!publicKey) {
+      throw new Error("Something is wrong with Account, no public key.");
+    }
+
+    // This is unlikely
+    if (!homeDomain) {
+      throw new Error("Something went wrong, home domain is not defined.");
+    }
+
+    log.instruction({ title: "Initiating a SEP-6 withdrawal" });
+
+    try {
+      // Check toml
+      const tomlResponse = await checkTomlForFields({
+        sepName: "SEP-6 withdrawal",
+        assetIssuer,
+        requiredKeys: [TomlFields.TRANSFER_SERVER],
+        networkUrl: networkConfig.url,
+        homeDomain,
+      });
+
+      // Check info
+      const infoData = await checkDepositWithdrawInfo({
+        type: CheckInfoType.WITHDRAWAL,
```

Comment:
💡  This name `CheckInfoType` isn't very self explanatory to me. Do you think we could rename it so its meaning becomes more explicit?